rocky/rocky_test.go

```diff
@@ -0,0 +1,86 @@
+package rocky_test
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/aquasecurity/vuln-list-update/rocky"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+)
+
+func Test_Update(t *testing.T) {
+	tests := []struct {
+		name               string
+		repomdFileName     string
+		updateInfoFileName string
+		expectedError      error
+	}{
+		{
+			name:               "happy path",
+			repomdFileName:     "testdata/fixtures/repomd_valid.xml",
+			updateInfoFileName: "testdata/fixtures/updateinfo_valid.xml.gz",
+			expectedError:      nil,
+		},
+		{
+			name:           "bad repomd response",
+			repomdFileName: "testdata/fixtures/repomd_invalid.xml",
+			expectedError:  xerrors.Errorf("failed to update security advisories of Rocky Linux 8 BaseOS x86_64: %w", errors.New("failed to fetch updateInfo path from repomd.xml")),
+		},
+		{
+			name:               "bad updateInfo response",
+			repomdFileName:     "testdata/fixtures/repomd_valid.xml",
+			updateInfoFileName: "testdata/fixtures/updateinfo_invalid.xml.gz",
+			expectedError:      xerrors.Errorf("failed to update security advisories of Rocky Linux 8 BaseOS x86_64: %w", errors.New("failed to fetch updateInfo")),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tsUpdateInfoURL := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				switch {
```

Comment:
Can we use `http.StripPrefix` and `http.FileServer`?
https://pkg.go.dev/net/http#StripPrefix

Reply:
I wrote it like this.
https://github.com/aquasecurity/vuln-list-update/pull/107/commits/38d634789e81b7b610d3cd399d21df6a019f9671#diff-a04921e003f983e8981667a23a94ceca9cbcece5302e9548b88a42d578b10a6cL46-R55

Reply:
refactor
https://github.com/aquasecurity/vuln-list-update/pull/107/commits/9f7e0b8ad82489c6394bb87b547fe810f1154393#diff-a04921e003f983e8981667a23a94ceca9cbcece5302e9548b88a42d578b10a6cR41